lib/rules/jsx-no-literals.js

```diff
@@ -17,17 +21,157 @@ const getText = require('../util/eslint').getText;
+    },
+  },
+  ignoreProps: {
+    type: 'boolean',
+  },
+  noAttributeStrings: {
+    type: 'boolean',
+  },
+};
+
+/**
+ * @typedef RawElementConfig
+ * @property {boolean} [noStrings]
+ * @property {string[]} [allowedStrings]
+ * @property {boolean} [ignoreProps]
+ * @property {boolean} [noAttributeStrings]
+ *
+ * @typedef RawOverrideOnlyConfig
+ * @property {boolean} [allowElement]
+ * @property {boolean} [applyToNestedElements=true]
+ *
+ * @typedef {RawElementConfig & RawOverrideOnlyConfig} RawOverrideConfig
+ *
+ * @typedef RawOverrideProperty
+ * @property {Record<string, RawOverrideConfig>} [elementOverrides]
+ *
+ * @typedef {RawElementConfig & RawOverrideProperty} RawConfig
+ *
+ * ----------------------------------------------------------------------
+ *
+ * @typedef ElementConfig
+ * @property {'element' | 'override'} type
+ * @property {boolean} noStrings
+ * @property {Set<string>} allowedStrings
+ * @property {boolean} ignoreProps
+ * @property {boolean} noAttributeStrings
+ *
+ * @typedef OverrideOnlyConfig
+ * @property {string} name
+ * @property {boolean} allowElement
+ * @property {boolean} applyToNestedElements
+ *
+ * @typedef {ElementConfig & OverrideOnlyConfig} OverrideConfig
+ *
+ * @typedef OverrideProperty
+ * @property {Record<string, OverrideConfig>} elementOverrides
+ *
+ * @typedef {ElementConfig & OverrideProperty} Config
+ *
+ * @typedef {Config | OverrideConfig} ResolvedConfig
+ */
+
+/**
+ * Normalizes the config and applies default values to all config options
+ * @param {RawConfig} config
+ * @param {boolean} [handleElementOverrides=true]
+ * @returns {Config}
+ */
+function normalizeConfig(config, handleElementOverrides) {
+  /** @type {Config} */
+  const normalizedConfig = {
+    type: 'element',
+    noStrings: config.noStrings || false,
+    allowedStrings: config.allowedStrings
+      ? new Set(map(iterFrom(config.allowedStrings), trimIfString))
+      : new Set(),
+    ignoreProps: config.ignoreProps || false,
+    noAttributeStrings: config.noAttributeStrings || false,
+    elementOverrides: {},
+  };
+
+  if (valueOrTrue(handleElementOverrides)) {
+    if (config.elementOverrides) {
+      normalizedConfig.elementOverrides = fromEntries(
+        reduce(
+          iterFrom(entries(config.elementOverrides)),
+          (acc, entry) => {
+            const key = entry[0];
+            const value = entry[1];
+
+            if (!(key && value)) return acc;
+            if (!reOverridableElement.test(key)) return acc;
+
+            acc.push([
+              key,
+              Object.assign(normalizeConfig(value, false), {
+                type: 'override',
+                name: key,
+                allowElement: value.allowElement || false,
+                applyToNestedElements: valueOrTrue(value.applyToNestedElements),
+              }),
+            ]);
+
+            return acc;
+          },
+          []
+        )
+      );
+    }
+  } else {
+    delete normalizedConfig.elementOverrides;
```

Comment:
it'd be better to not add it, rather than conditionally delete it, i think?

Reply:
Typescript was being a pain about this. But I can probably refactor the function
a bit to make it get along.